recipes/zbar/all/conanfile.py

```diff
@@ -87,23 +85,25 @@ def build_requirements(self):
     def validate(self):
         if self.settings.os == "Windows":
             raise ConanInvalidConfiguration("Zbar can't be built on Windows")
-        if tools.is_apple_os(self.settings.os) and not self.options.shared:
+        if is_apple_os(self) and not self.options.shared:
             raise ConanInvalidConfiguration("Zbar can't be built static on macOS")
         if self.options.with_xv:            #TODO add when available
             self.output.warn("There is no Xvideo package available on Conan (yet). This recipe will use the one present on the system (if available).")
-        if tools.Version(self.version) >= "0.22" and cross_building(self):
+        if Version(self.version) >= "0.22" and cross_building(self):
             raise ConanInvalidConfiguration("{} can't be built on cross building environment currently because autopoint(part of gettext) doesn't execute correctly.".format(self.name))
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        tc.generate()
 
     def _configure_autotools(self):
         if not self._autotools:
-            if tools.Version(self.version) >= "0.22":
-                with tools.chdir(self._source_subfolder):
-                    self.run("autoreconf -fiv")
-            self._autotools = AutoToolsBuildEnvironment(self)
+            self._autotools = Autotools(self)
```

Comment:
You should no longer cache the the build helpers, you can check the package template for more details

Reply:
You can check the template for the updates https://github.com/conan-io/conan-center-index/blob/7ce55ac4cb6d1f53ffaca6bb767e6f25655c4b67/docs/package_templates/autotools_package/all/conanfile.py#L150

:) Thanks for updating this recipe